src/components/TodoItem.tsx

```diff
@@ -0,0 +1,176 @@
+}
+
+interface IBtn {
+    done: boolean;
+    edit: boolean;
+}
+
+const TodoItem = ({ item }: ITodoItemProps) => {
+    const { id, content, done } = item;
+    const { deleteItem, toggleItem, editItem } = useTodoContext();
+
+    const [edit, setEdit] = useState(false);
+    const { value, onChange, onSubmit } = useForm(item.content, editItem, item.id);
+
+    const inputRef = useRef<HTMLInputElement>(null);
+    const editFormRef = useRef<HTMLFormElement>(null);
+
+    //참고3 : editForm 외의 부분을 클릭했을 때 setEdit(false)
+    useEffect(() => {
+        const handleClickOutside = (e: MouseEvent): void => {
+            if (editFormRef.current && !editFormRef.current.contains(e.target as Node)) {
+                setEdit(false);
+            }
+        };
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [inputRef]);
+
+    // 참고2
+    useEffect(() => {
+        const { current } = inputRef;
+        if (current !== null) {
+            current.focus();
+        }
+    }, [edit]);
+
+    const handleEditBtn = () => {
+        setEdit(true);
+    };
+
+    return (
+        <Item>
+            {edit ? (
+                <ItemEditForm
+                    onSubmit={(e) => {
+                        onSubmit(e);
+                        setEdit(false);
+                    }}
+                    ref={editFormRef}
+                >
+                    <InputEdit value={value} onChange={onChange} ref={inputRef} />
+                    <StyledSubmitEdit type="submit">
+                        <Send />
+                    </StyledSubmitEdit>
+                </ItemEditForm>
+            ) : (
+                <ItemContentBox
+                    onClick={() => {
+                        toggleItem(id);
+                    }}
+                >
+                    <RadioButton done={done} edit={false}></RadioButton>
+                    <ItemContent>{content}</ItemContent>
+                </ItemContentBox>
+            )}
+            <ItemContentBox>
+                {edit ? (
+                    ''
+                ) : (
+                    <>
+                        <StyledEdit onClick={handleEditBtn} />
+                        <StyledRemove onClick={() => deleteItem(item.id)} />
+                    </>
+                )}
+            </ItemContentBox>
+        </Item>
+    );
+};
+
+export default TodoItem;
+
+const Item = styled.li`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    height: 35px;
+`;
+
+const ItemContentBox = styled.div`
+    display: flex;
+    align-items: center;
+`;
+
+// 참고1
+const RadioButton = styled.div<IBtn>`
+    width: 16px;
+    height: 16px;
+    margin-right: 10px;
+    border-radius: 50%;
+    border: 2px solid #8989bb;
+    cursor: pointer;
+    background-color: ${(props: IBtn) => (props.done ? '#8989bb' : 'none')};
```

Comment:
```suggestion
    background-color: ${({ done }) => (done ? '#8989bb' : 'none')};
```
이렇게 destructuring 해줘도 좋을 것 같아요!